File: code/python/reverseStr.py
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        if len(s) < k:
            return s[::-1]
        if len(s) >= k and len(s) < 2* k:
            tmp = s[:k]
            tmp = tmp[::-1]
            return tmp + s[k:]

        tmp = s[:k]
        tmp = tmp[::-1]
        return tmp + s[k:2*k] + self.reverseStr(s[2*k:],k)
            
        

File: code/java/reverseStr.java
class Solution {
    public String reverseStr(String s, int k) {
        if(s.length() < k)
            return reverse(s);
        else if(s.length()>=k && s.length() < 2* k){
            return reverse(s.substring(0,k)) + s.substring(k,s.length());
        }
        else{
            return reverse(s.substring(0,k)) + s.substring(k,2*k) 
                + reverseStr(s.substring(2*k,s.length()),k);
        }
    }
    
    public String reverse(String s){
        if(s.length() == 0)
            return s;
        String res = "";
        for(int i=s.length();i>0;i--)
            res = res + s.substring(i-1,i);
        return res;
    }
}
